### ISSUE TITLE
False negative when using a TypeVar that is not bound to scope

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#5710]
In most cases, pyright emits an error when a TypeVar is referenced in a location where it has no meaning, but the following code demonstrates a case where this is not properly reported.

```python
from typing import Generic, TypeVar

T = TypeVar("T")

class A(Generic[T]):
    pass

A[T]()
```

This is related to mypy issue: https://github.com/python/mypy/issues/12948

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.323, which I just published. It will also be included in a future release of pylance.
